pkg/node/address_linux.go

```diff
@@ -57,16 +61,19 @@ retryScope:
 				continue
 			}
 			if len(a.IP) >= ipLen {
-				if ip.IsPublicAddr(a.IP) {
-					ipsPublic = append(ipsPublic, a.IP)
-				} else {
-					ipsPrivate = append(ipsPrivate, a.IP)
-				}
 				// If the IP is the same as the preferredIP, that
 				// means that maybe it is restored from node_config.h,
 				// so if it is present we prefer this one.
 				if a.IP.Equal(preferredIP) {
 					hasPreferred = true
+				} else if a.Flags&unix.IFA_F_SECONDARY > 0 {
+					// Skip secondary addresses
+					continue
```

Comment:
AFAICS, there are two independent changes here: the sort and the filter for secondary addresses. In your commit you mention that this fixes an issue with floating IPs in your case. Which change is fixing your issue exactly?

Given they are independent changes, I would also prefer to have them in separate commits. We may need to backport one change without backporting the other.

Reply:
> The sort by creation time is enough to fix our issue, because the floating IP is assigned by an agent on the node after the primary IP has been assigned.

That's what I don't understand. If the floating IP is assigned to the same interface as the primary address, why does sorting by interface indexes change anything? 

> I'll separate them into separate commits.

Thanks!

Reply:
Ah, I see that makes sense 👍 

Yep, an explanation of that in the commit description would be great!